site/gdocs/ArticleBlock.tsx

```diff
@@ -492,6 +493,15 @@ export default function ArticleBlock({
                 className={getLayout("additional-charts", containerType)}
             />
         ))
+        .with({ type: "expandable-paragraph" }, (block) => (
+            <ExpandableParagraph
+                className={getLayout("expandable-paragraph", containerType)}
+            >
+                {block.items.map((item, i) => (
```

Comment:
Hmm. This is effectively a new `containerType` which needs to be accounted for. Prominent links are _kind of_ working correctly by accident, though there's a CSS rule `:nth-child(n + 3)` defined in `expandable-paragraph.scss` which is breaking things depending on how many elements we have in the expandable-paragraph.

The wrinkle is that this is a container which could plausibly be nested inside another container (e.g. inside a sticky-left-left-column) and the current pattern doesn't really handle that.

We would need to create a new set of layouts for `sticky-right-left-column-expandable-paragraph`

Take a look at the following Archie to see what I mean:

```
[+body]

[.+expandable-paragraph]
One
Two
Three

{.prominent-link}
title: I am a prominent link
url: https://ourworldindata.org
thumbnail: checkers_2_1.webp
{}

[]

{.sticky-right}
[.+left]

I am a sticky left column

{.prominent-link}
title: I am a prominent link outside of the expandable paragraph
url: https://ourworldindata.org
thumbnail: checkers_2_1.webp
{}

[.+expandable-paragraph]
I am expandable paragraph inside a sticky left column

{.prominent-link}
title: I am a prominent link inside the expandable paragraph
url: https://ourworldindata.org
thumbnail: checkers_2_1.webp
{}

[]


[]
[.+right]

I am a sticky right column

{.prominent-link}
title: I am a prominent link outside of the expandable paragraph
url: https://ourworldindata.org
thumbnail: checkers_2_1.webp
{}

[.+expandable-paragraph]
I am expandable paragraph inside a sticky right column

{.prominent-link}
title: I am a prominent link inside of the expandable paragraph
url: https://ourworldindata.org
thumbnail: checkers_2_1.webp
{}

[]

[]
{}
[]
```

![expandable-paragraph grid prominent link demo](https://github.com/owid/owid-grapher/assets/11844404/fe114c29-7a86-4520-9782-218a126e8bc2)

Looks like there's already a bug with sticky-left-left-column prominent links not having a correct layout defined. The outlined prominent link is the only one that's rendering correctly! 😩 

So... take a look at let me know what you think 🙃

Reply:
We might be able to leverage the `inherit` keyword with a couple of specific `grid` CSS properties to avoid having to define each permutation so that the content div just picks up its parent's size.

Reply:
@ikesau both fixes are in:

- alignment of prominent links with the grid when inside a sticky-left or sticky-right layout (in either columns).
<img width="1440" alt="Screenshot 2023-05-29 at 15 20 13" src="https://github.com/owid/owid-grapher/assets/13406362/ae9de2dc-5565-4e7a-a406-b878312728b5">

- expand behaviour of `ExpandableParagraph` now doesn't rely on changing the `display` property, and uses instead the react-animate-height package.

https://github.com/owid/owid-grapher/assets/13406362/2c0affc0-c526-40b9-9154-8cc47bb9f0c5


- [datapage preview](http://staging-site-2204-datapages-add-collapse-functionality-to-faq-s/admin/grapher/banana-production)
- [regular gdoc preview](http://staging-site-2204-datapages-add-collapse-functionality-to-faq-s/admin/gdocs/1fLhcFxbWSa09oM6ubryg5HcaZ4N3xsT44OKW0_7tAJY/preview)